docs/interface-layers.rst

```diff
@@ -0,0 +1,16 @@
+Reactive Charms and Interface Layers
+------------------------------------
+
+Reactive charms work best as a part of layered charms, and the reactive pattern
+enables the use of interface layers which encapsulate the communication protocol
+over a Juju interface when two applications are related together.  By
+encapsulating this protocol, it ensures that two applications that support the
+same interface will always be able to speak to each other correctly.  It also
+provides a well-defined and documented API for the charm author to use, without
+having to get mired in the details of relation keys, data encoding, or what
+order the communication has to proceed in.
+
```

Comment:
It might be nice to add an ascii diagram here to visualize how the layers work together.